Fixing missing dependencies for Gradle projects with "azure-function-http" feature

wetted opened the pull request:
Fixing missing dependencies for Gradle projects with "azure-function-http" feature, to make that consistent with Maven projects. Refactor dependency logic from Rocker files to Feature classes.

wetted committed:
Fixing missing dependencies for Gradle projects with "azure-function-http" feature, to make constant with Maven projects. Refactor dependency logic from Rocker files to Feature classes.

wetted committed:
fic checkstyle error

wetted commented:
@sdelamo if #1709 is merged first – has a bugfix I need – I can add a minor improvement to `AzureHttpFunctionSpec`.

wetted committed:
Add `micronaut-azure-function` only for FUNCTION app type.

sdelamo reviewed (changes requested):
> Fixing missing dependencies for Gradle projects with "azure-function-http" feature, to make that consistent with Maven projects. Refactor dependency logic from Rocker files to Feature classes.

It is not a bug in gradle. The [gradle plugin adds those dependencies automatically.](https://github.com/micronaut-projects/micronaut-gradle-plugin/blob/3.7.x/minimal-plugin/src/main/java/io/micronaut/gradle/MicronautRuntimeDependencies.java).

sdelamo committed:
Merge branch '3.9.x' into azure_function_http_fix

sdelamo committed:
Merge branch '3.9.x' into azure_function_http_fix

sdelamo committed:
add dependencies only for maven

Gralde adds those dependencies automatically https://github.com/micronaut-projects/micronaut-gradle-plugin/blob/3.7.x/minimal-plugin/src/main/java/io/micronaut/gradle/MicronautRuntimeDependencies.java

sdelamo commented on starter-core/src/test/groovy/io/micronaut/starter/feature/function/azure/AzureHttpFunctionSpec.groovy:
```diff
@@ -58,4 +62,21 @@ class AzureHttpFunctionSpec extends BeanContextSpec  implements CommandOutputFix
+    void 'test gradle azure-function-http feature for language=#language and buildTool=#buildTool'() {
+        when:
+        String template = new BuildBuilder(beanContext, buildTool)
+                .features(['azure-function-http'])
```
azure-function-http is invisible. We should test `azure-function` and `ApplicationType.DEFAULT` and  `azure-function` and `ApplicationType.FUNCTION`

sdelamo committed:
set previous behaviour
